src/components/Textarea/Textarea.tsx

```diff
@@ -99,6 +105,28 @@ const Textarea = ({
   const defaultTextAreaId = useId();
   const textAreaId = id || defaultTextAreaId;
 
+  const handleKeyDown = useCallback(
+    (event: KeyboardEvent) => {
+      const textareaElement = document.getElementById(textAreaId);
+      if (
+        event.key === "Enter" &&
+        (event.ctrlKey || event.metaKey) &&
+        document.activeElement === textareaElement
```

Comment:
I could be misunderstanding something, but I'd expect this focus check to be done against a ref for the textArea, rather than having to perform a getElementById check

Reply:
I actually did attempt with a ref at first because I wanted to use it alongside .hasFocus(), but it threw up some problems. Perhaps I used it incorrectly. Any suggestions there?